packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts

```diff
@@ -65,13 +66,20 @@ export class CrossDomainFinalizer {
     let nonce = await this.l2Client.l2Web3.eth.getTransactionCount(this.account);
     for (const l2Token of bridgeableL2Tokens) {
       // For each bridgeable L2Token, check the balance in the deposit box. If it is greater than
-      // crossDomainFinalizationThreshold, as a percentage, then we can bridge it.
+      // crossDomainFinalizationThreshold, as a percentage, then we can bridge it. If the liquidity utilization is
+      // greater than 75% then half the crossDomainFinalizationThreshold to send funds more aggressively over the
+      // bridge at high utilizations.
 
       try {
         const { symbol, decimals, l2PoolBalance } = await this._getL2TokenInfo(l2Token);
         const l1PoolReserves = await this._getL1PoolReserves(l2Token);
+        const l1PoolUtilization = await this._getL1PoolUtilization(l2Token);
 
-        if (l2PoolBalance.gt(toBN(this.crossDomainFinalizationThreshold).mul(l1PoolReserves).div(toBN(100)))) {
+        const scaledCrossDomainFinalizationThreshold = l1PoolUtilization.gt(toBNWei("0.75"))
```

Comment:
I'm curious why you slash the `crossDomainFinalizationThreshold` figure instead of just using the utilization threshold to determine whether to bridge. For example, could you not just set a rule: if utilization > 75% OR L2PoolBalance > (`scaledCrossDomainFinalizationThreshold * l1PoolReserves`) then bridge

Reply:
because you don't always want to bridge if we are above 75%. this is an _and_ condition, not an _or_ condition. i.e if we are at 76% utilization and the bridge pool has 10 wei in it we don't want to bridge. briding is expensive so should not always be done but rather be done _more_ quickly.